src/algebra/algebra/basic.lean

```diff
@@ -581,6 +582,26 @@ lemma map_list_prod (s : list A) :
   φ s.prod = (s.map φ).prod :=
 φ.to_ring_hom.map_list_prod s
 
+section prod
+
+/-- First projection as `alg_hom`. -/
+def fst : A × B →ₐ[R] A :=
+{ commutes' := λ r, rfl, .. ring_hom.fst A B}
+
+/-- Second projection as `alg_hom`. -/
+def snd : A × B →ₐ[R] B :=
+{ commutes' := λ r, rfl, .. ring_hom.snd A B}
+
+lemma _root_.linear_map.inl_map_mul (a₁ a₂ : A) : linear_map.inl R A B (a₁ * a₂) =
+  linear_map.inl R A B a₁ * linear_map.inl R A B a₂ :=
+prod.ext rfl (by simp)
+
+lemma _root_.linear_map.inr_map_mul (b₁ b₂ : B) : linear_map.inr R A B (b₁ * b₂) =
+  linear_map.inr R A B b₁ * linear_map.inr R A B b₂ :=
```

Comment:
This is more generally true than for `comm_semiring R` `algebra A` right? `semiring R` `module R A` `non_unital_non_assoc_semiring A` feels like it should suffice

Reply:
Suggesting it belongs in linear_algebra/prod.lean

Reply:
Yes, it's true essentially everywhere. Let me experiment with the brand new `non_unital_non_assoc_semiring A`.